src/clzones.cpp

```diff
@@ -827,7 +827,11 @@ cata::optional<tripoint> zone_manager::get_nearest( const zone_type_id &type, co
 zone_type_id zone_manager::get_near_zone_type_for_item( const item &it,
         const tripoint &where, int range ) const
 {
-    auto cat = it.get_category();
+    const item_category &cat = it.get_category();
+    if( cat.zone() ) {
+        return *cat.zone();
+    }
```

Comment:
This should be moved after line 850 to preserve the original order of the if conditions.